### DISCUSSION TITLE
Option to only trigger LSP on write (or other "events")

### DISCUSSION THREAD

**Discussion by bczsalba** [helix-editor/helix#9137] [General]
I spent some time on trying to figure out a way to only get LSP feedback after the file was written to, and couldn't come up with any results, so if I just overlooked something please let me know.

I come from a pretty vanilla vim config, so getting constantly reminded by issues in my code (which I'm on my way to fix) has been very distracting.

Maybe some option like `lsp-trigger = ["write", "insert"]` could fix this? Not sure how easy that would be to implement, since I haven't really looked at the code much yet.

**Comment by yowayb** [comment]
Would this depend on #8021 ?

**Comment by pascalkuthe** [comment]
In theory that could help with implementing this feature but in practice this is just not going to work. This was discussed on Matrix quite a while back. The LSP spec is not designed for something like this and it would require a lot of hacks to make work (and disable all other lsp features like autocompletions in the process). I don't think this is something that should be added to core
